\chapter{Introduction}

Nearly half of the world's population are at risk of contracting malaria.  In 2015, 214 million malaria infections, resulting in 438,000 deaths, were reported. Nearly 90\% of the reported infections and deaths have occurred in Sub-Saharan Africa.  Mosquitoes in the \emph{Anopheles} genus are the sole vectors of the malaria-causing \emph{Plasmodium} parasites \cite{Neafsey2015,Neafsey2010,Lawniczak2010}, and as such, efforts to control \emph{Anopheles} populations are key to malaria eradication efforts \cite{Holt2002}.

\emph{Anopheles} species have been subject to evolutionary adapations due to host-parasite interactions and specializations needed to thrive when living among and feed on humans \cite{Neafsey2015}. Population control efforts are informed through studies of insect biology and characterizerations of these adaptations, which are enabled by sequencing and comparative analysis of insect genomes.   Such studies have been fruitful, with results including the identification of chemosensory receptors responsible for detection of human hosts \textcolor{red}{CITE} and the molecular bases of variations in insecticide resistance between populations and species \cite{Lawniczak2010}.

Bioinformatic analysis of insect genomes is complicated by unique characteristics of insect biology.  For example, mammalian chemosensory receptors are G Protein Coupled Receptors (GPCRs), while insect chemosensory receptors are ion channels which are poorly-conserved on the sequence level.  Insect genomes tend to have low linkage disequilibrium (LD), or correlation  between variations located nearby on chromosomes, resulting in low statistical power when identifying interesting variations.  And, insect genomes tend to undergo relatively rapid evolution, making it difficult to find appropriate reference genomes to aid assembly.

The larger bioinformatics community is adept at developing algorithms and accompanying software to meet such challenges.  However, constantly developing new tools is inefficient and can inhibit research progress.  Method development is time consuming and requires significant resources in terms of man power.  The larger scientific community can be slow to trust new methods, inhibiting the usage of newer, more accurate tools.

The larger scientific community has good reason to be skeptical of new methods and software packages. It can take years to identify all of the bugs in software and ensure that the results are correct.  With high turn over due to graduating students and relatively short funding cycles, research groups rarely have the resources to maintain software implementations beyond an initial proof of concept, leading to a flood of new tools without adequate testing which are poorly maintained.

Development time and number of bugs can be reduced through writing \emph{less new code} and and re-using more existing libraries. The machine learning community has invested significant effort into developing mature, standardized libraries for a variety of languages and platforms.  Due to their large user bases and maturity, most bugs have identified and fixed and offer optimal performance.  Actively supported by communities of developers, new methods are added reasonably quickly and users can expect continued support.  Some of these libraries are able to utilize GPUs and distributed computing to scale to massive data volumes.

\section{Contributions}
The key challenge is to demonstrate that standard machine learning techniques can be adapted to overcome the challenges facing bioinformatic analysis of insect vector genomes in place of more specialized methods.  In this dissertation, we demonstrate exactly that.

%In this thesis, we perform bioinformatic analyses of several insect genomes. Throughout the analysis process, we use a combination of existing tools and create new tools where necessary to improve upon the analysis of insect genomes in the face of their unique challenges.  As part of our work, we developing new tools specific to the study of insects.

\section{Synopsis}
In this document, we

\begin{enumerate}
\item Perform a comparative analysis of the recently-sequenced genomes of two sand fly species, \emph{L. longipalpis} and \emph{P. papatasi}. (Chapter 2)
\item Evaluate existing Hidden Markov Model (HMM)-based tools for identifying G Protein-Coupled Receptors (GPCRs) on insect vector genomes and demonstrate how the simple approach of creating an ensemble of existing classifiers can improve accuracy by at least 5-11\%. (Chapter 3)
\item Apply Random Forests to ranking SNPs from incipient species of insect vectors.  We identify multiple sources of bias that can arise and provide solutions that utilize ``out-of-the-box'' implementations of decision trees.  We provide a software implementation of our workflow using the popular machine learning library sci-kit learn and apply it to studying SNPs of \emph{Anopheles} mosquitoes. (Chapter 4)
\end{enumerate}

We conclude by identifying topics for further investigation.
