\chapter{Introduction}

Arthropod vectors are increasingly becoming important in emerging human disease outbreaks.  In addition to the recent Zika outbreak vectored by the mosquito \emph{Aedes aegypti}, linked to serious medical outcomes, nearly half of the world's population is still at risk of contracting malaria.  In 2015, 214 million malaria infections, resulting in 438,000 deaths, were reported. Nearly 90\% of the reported infections and deaths have occurred in Sub-Saharan Africa.  Mosquitoes in the \emph{Anopheles} genus are the sole vectors of the malaria-causing \emph{Plasmodium} parasites \cite{Neafsey2015,Neafsey2010,Lawniczak2010}. As such, efforts to control populations of insect vectors such as \emph{Anopheles gambiae} are key to efforts to eradicate malaria and other insect-borne diseases \cite{Holt2002}.

\emph{Anopheles} species have been subject to evolutionary adapations due to host-parasite interactions and specializations needed to thrive when living among and feed on humans \cite{Neafsey2015}. Population control efforts are informed through studies of insect biology and characterizerations of these adaptations, which are enabled by sequencing and comparative analysis of insect genomes.   Such studies have been fruitful, with results including the identification of chemosensory receptors responsible for detection of human hosts \textcolor{red}{CITE} and the molecular bases of variations in insecticide resistance between populations and species \cite{Lawniczak2010}.

Bioinformatic analysis of insect genomes is complicated by unique characteristics of insect biology.  For example, mammalian chemosensory receptors are G Protein Coupled Receptors (GPCRs), while insect chemosensory receptors are ion channels which are poorly-conserved on the sequence level.  Insect genomes tend to have low linkage disequilibrium (LD), or correlation  between variations located nearby on chromosomes, resulting in low statistical power when identifying interesting variations.

The larger bioinformatics community is adept at developing algorithms and accompanying software to meet such challenges.  Developing methods and software is expensive in terms of time and resources (human, financial), however. A majority of bioinformatics software is developed for a single or small number of analyses, abandoned once the analyses are done. Studies of defect rates in software have found that bugs tend to most frequently occur in files with the most recent activity \textcolor{red}{cite}.  Interpreted another way, new bugs can be introduced in the process of adding new features and fixing previously-known bugs can; over time, however, modules mature and the rate at which new bugs are found declines, leading to reliable components. Furthermore, bioinformatics applications often employ specalized methods not available in common libraries.  As such, bioinformatics software often consists of completely custom implementations of every component from file parsers to complex algorithms and statistical tests.  We can conclude that most bioinformatics applications consist of new, custom code with high defect rates, leading to significant development costs due to time spent debugging.

Furthermore, bioinformatics software is often used on large data sets and employs computationally-expensive algorithms.  Since reducing software run times from hours or days to minutes can significantly boost productivity, enabling a much shorter loop in idea-implementation-evaluation cycles, it is highly desireable to optimize and parallelize code.  Low-level languages like C are often used to achieve optimal usage of hardware. C is more error-prone than high-level langauges like Python, however, due to manual memory management and poor type safety.  Parallelization is also known to increase software complexity and defect rates, especially since race conditions are notoriously difficult to debug. Both requirements lead to increased time and effort spent debugging and validating software.

Machine learning libraries offer an attractive alternative.  Due to general applicability, machine learning techniques are widely applied through academia and industry.  A number of mature and widely-used machine learning libraries exist including the scikit-learn \cite{scikit-learn} library for Python and the MLLib library bundled with the Apache Spark distributed-computing engine.  As of the time this was written, scikit-learn and Apache Spark have been around for 8 and 7 years, over 550 and 1,000 contributors, and over 2,600 and 1,100 citations, respectively. Both libraries implement a number of popular machine learning techniques including Random Forests, K-Means clustering, Support Vector Machines (SVMs), and Naive Bayes classifiers. 

Scikit-learn and MLLib offer best-in-class performance and scalability.  Scikit-learn is optimized via C and Fortran implementations of performance-critical sections and can efficiently scale through support for parallel processing.  Through Apache Spark and conscientious design, MLLib can scale from a single machine to a large cluster to support data sets too large to fit into memory and algorithms too slow to use productively when running on a single CPU core. Users of scikit-learn and MLLib reap the resulting performance and scalability benefits essentially for free; the libraries largely hide the details from the user.

By adopting machine learning techniques as building blocks for developing bioinformatics methods and applications, machine learning libraries could be adopted, benefitting method developers. Software defects tend to be over-represented in functions employing complex algorithms, parallelization, and low-level optimizations like manual memory management, exactly the functionality that would be replaced by the machine learning libraries, presumably decreasing defect rates and time and labor spent debugging. Additionally, scaling software to large data sets and achieving optimal hardware utilization could be accomplished with far less effort; the optimizations and parallelization techniques employed by machine learning libraries are hidden completely from the user or accomplished by through abstractions that are easier to use.

\section{Contributions}
Prerequisite is to demonstrate that machine learning techniques are applicable to problems in bioinformatics and just as, if not more, effective when replacing or augmenting specialized techniques.  In this dissertation, we consider several of the unique challenges facing bioinformatics of insect genomes and apply machine learning to two common problems in bioinformatics, finding members of a gene family in genomes and ranking genetic variants based on association with population structures.  We demonstrate that both problems are instances of classic machine learning problems, classification and feature selection, respectively.

First, we describe analyzes of the genomes and RNASeq expression of genes from two sand fly species, \emph{Lutzomyia longipalpis} and \emph{Phlebotomus papatasi}, which are vectors of leishmaniasis. These analyses provide background into the types of questions that biologists seek to answer and the challenges associated with insect genome bioinformatics. Analysis of the size of the scaffolds and distribution of genes across the scaffolds indicated that the genome assemblies were highly fragmented.  We found little evidence of macrosynteny, likely due to a combination of the fragmentation of the assembly and rapid rearrangement of genes insects.  We, did, however, find several examples of microsynteny, mostly of bookkeeping genes that are present in most organisms and known to be highly conserved.  A microsynteny block of Niemann-Pick like genes found spanning both sand fly species and \emph{An. gambie} was of particular interest; the genes were not in synteny with their orthologs in the model organism \emph{D. melanogaster}, despite the role of the fruit fly as a model for the study of mutations causing Niemann-Pick disease in humans. Analysis of the distribution of rates of non-silent mutations to silent mutations (dN/dS) across single-copy orthologs indicated a noticeably lower rate when comparing the sand flies with each other versus \emph{An. gambiae} or \emph{D. melanogaster}.    The shift in the dN/dS distribution is likely caused by assembly issues -- in particular, we have observed that gene models are truncated such that conserved regions are intact but less conserved regions are missing. \textcolor{red}{RNASeq}

Secondly, we describe a novel ensemble approach for identifying G Protein-Coupled Receptors (GPCRs) in insect vector genomes.  Existing GPCR classifiers such as GPCRHMM and PredCouple present several limitations: (1) they are trained using GPCRs from a diverse set of organisms including human (\emph{Homo homo sapiens}) and the fruit fly\emph{Drosophila melanogaster}, resulting in sub-optimal performance when applied to insect vectors and (2) their resulting scores can be difficult to interpret.  We describe a novel pipeline, called Ensemble*, that utilizes GPCRHMM and Pfam GPCR Hidden Markov Models (HMMs) but improves sensitivity, accuracy, and interpretability over GPCRHMM and Pfam HMMs alone. For each classifier, Ensemble* uses an empirical conditional probability function, trained on known GPCRs from the mosquitoes \emph{Aedes aegypti}, \emph{Anopheles gambiae}, the louse \emph {Pediculus humanus}, the honey bee \emph{Apis mellifera}, \emph{Drosophila melanogaster}, and \emph{Homo homo sapiens}, to estimate the probability that a protein is a GPCR given a score from the classifier.  An ensemble was formed from the classifiers by using a weighted average the classifiers' likelihood functions.  We evaluated the sensitivity and accuracy of Ensemble*, demonstrating improved sensitivities of up to 10\% on \emph{Aedes aegypti}, \emph{Anopheles gambiae}, \emph{Apis mellifera}, and \emph{Pediculus humanus} with no loss of sensitivity for \emph{Drosophila melanogaster} and \emph{Homo homo sapiens}. Using Ensemble*, we re-analyzed the GPCR repoirteres of the vectors \emph{Aedes aegypti}, \emph{Anopheles gambiae}, resulting in the identification of 30 additional putative GPCRs, of which 19 were validated and confirmed via sequence similarity to known GPCRs with BLAST.

Lastly, we describe a workflow for ranking SNPs from incipient \emph{Anopheles} species using Random Forests.  Along the way, we demonstrate several causes of bias using numerical experiments and describe appropriate solutions that can be used with standard, \emph{unmodified} implementations of decision trees available in popular machine learning libraries. We provide a software package called Asaph that implements the workflow using Python, numpy, scipy, and scikit-learn.  Asaph is applied to SNPs from \emph{Anopheles gambiae}, \emph{Anopheles coluzzii}, and two forms of \emph{Anopeheles funestus}, Folonzo and Kiribina, resulting in the identification of SNPs in genes previously reportedly to be associated with differences in insecticide resistance between the populations.

\textcolor{red}{sand flies -$>$ challenges in insect bioinformatics}

\textcolor{red}{ensemble* -$>$ existing bioinformatics tools can be adapted and improved for use on insect genomes by employing techniques from machine learning (ensembles) and statistics (conditional probabilities)}

\textcolor{red}{Asaph -$>$ more common machine learning techniques (e.g., Random Forests) can be used to replace more specialized bioinformatics techniques, allowing for the use of existing libraries and their advantages}

\textcolor{red}{draw conclusion, bringing argument back to original goal}

\section{Synopsis}
The rest of this dissertation is organized as follows:

\begin{itemize}
\item Chapter 2: Comparative analysis of the genomes of and RNASeq expression of genes from two sand fly species, \emph{L. longipalpis} and \emph{P. papatasi}.
\item Chapter 3: Evaluation of Hidden Markov Model (HMM)-based tools for identifying G Protein-Coupled Receptors (GPCRs) in insect vector genomes and demonstratation of how a simple approach for creating an ensemble of existing classifiers can improve accuracy by at least 5-11\%.
\item Chapter 4: Application of Random Forests to ranking SNPs from incipient species of \emph{Anopheles} mosquitoes.  We identify multiple sources of bias that can arise and provide solutions that utilize ``out-of-the-box'' implementations of decision trees.  We provide a software implementation of our workflow using the popular machine learning library sci-kit learn, which we apply to the study of SNPs from several \emph{Anopheles} species.
\end{itemize}

We conclude by identifying topics for further investigation. \textcolor{red}{expand}
