\chapter{Introduction}

Arthropod vectors are increasingly becoming important in emerging human disease outbreaks.  In addition to the recent Zika outbreak vectored by the mosquito \emph{Aedes aegypti}, linked to serious medical outcomes, nearly half of the world's population is still at risk of contracting malaria.  In 2015, 214 million malaria infections, resulting in 438,000 deaths, were reported. Nearly 90\% of the reported infections and deaths have occurred in Sub-Saharan Africa.  Mosquitoes in the \emph{Anopheles} genus are the sole vectors of the malaria-causing \emph{Plasmodium} parasites \cite{Neafsey2015,Neafsey2010,Lawniczak2010}. As such, efforts to control populations of insect vectors such as \emph{Anopheles gambiae} are key to efforts to eradicate malaria and other insect-borne diseases \cite{Holt2002}.

\emph{Anopheles} species have been subject to evolutionary adapations due to host-parasite interactions and specializations needed to thrive when living among and feed on humans \cite{Neafsey2015}. Population control efforts are informed through studies of insect biology and characterizerations of these adaptations, which are enabled by sequencing and comparative analysis of insect genomes.   Such studies have been fruitful, with results including the identification of chemosensory receptors responsible for detection of human hosts \textcolor{red}{CITE} and the molecular bases of variations in insecticide resistance between populations and species \cite{Lawniczak2010}.

Bioinformatic analysis of insect genomes is complicated by unique characteristics of insect biology.  For example, mammalian chemosensory receptors are G Protein Coupled Receptors (GPCRs), while insect chemosensory receptors are ion channels which are poorly-conserved on the sequence level.  Insect genomes tend to have low linkage disequilibrium (LD), or correlation  between variations located nearby on chromosomes, resulting in low statistical power when identifying interesting variations.

The larger bioinformatics community is adept at developing algorithms and accompanying software to meet such challenges.  However, constantly developing new tools is inefficient and can inhibit research progress.  Method development is time consuming and requires significant resources.

Development time and the number of bugs can be reduced through writing \emph{less new code} and and making more extensive usage of existing libraries. The machine learning community has invested significant effort into developing and standardizing around libraries for a variety of languages and platforms.  Libraries such as scikit-learn \cite{scikit-learn} have large users bases and have gained the respect of the scientific community for their quality, performance, active community of developers, and maturity. New algorithms are implemented reasonably quickly, releases occur on a regular schedule, and users can expect continued support.  Some of these libraries \textcolor{red}{CITE} are able to utilize GPUs and distributed computing to scale to massive data volumes.

\section{Contributions}
The key challenge is to demonstrate that standard machine learning techniques can be adapted to overcome the challenges facing bioinformatic analysis of insect vector genomes in place of more specialized methods.  In this dissertation, we demonstrate the role of machine learning methods in insect bioinformatics. 

We begin by analyzing the genomes and RNASeq expression of genes from two sand fly species, \emph{Lutzomyia longipalpis} and \emph{Phlebotomus papatasi}, which are vectors of leishmaniasis. \textcolor{red}{more details on results} These analyses provide background into the types of questions that biologists seek to answer and the challenges associated with insect genome bioinformatics.

Secondly, we evaluate methods for identifying G Protein-Coupled Receptors (GPCRs) for their sensitivity and accuracy when applied to insect genomes.  We use our evalution to design an ensemble that combines GPCRHMM and Pfam HMMs with empirical likelihood functions to identify and rank potential GPCRs.

Lastly, we describe a workflow for ranking SNPs from incipient \emph{Anopheles} species using Random Forests.  Along the way, we demonstrate the presence of bias using numerical experiments and describe appropriate solutions that can be used with standard, \emph{unmodified} implementations of decision trees available in popular machine learning libraries. We provide a software package called Asaph that implements the workflow using Python, numpy, scipy, and scikit-learn.  Asaph is applied to SNPs from \emph{Anopheles gambiae}, \emph{Anopheles coluzzii}, and two forms of \emph{Anopeheles funestus}, Folonzo and Kiribina, resulting in the identification of SNPs in genes previously reportedly to be associated with differences in insecticide resistance between the populations.

\textcolor{red}{draw conclusion, bringing argument back to original goal}

\section{Synopsis}
The rest of this dissertation is organized as follows:

\begin{itemize}
\item Chapter 2: Comparative analysis of the genomes of and RNASeq expression of genes from two sand fly species, \emph{L. longipalpis} and \emph{P. papatasi}.
\item Chapter 3: Evaluation of Hidden Markov Model (HMM)-based tools for identifying G Protein-Coupled Receptors (GPCRs) in insect vector genomes and demonstratation of how a simple approach for creating an ensemble of existing classifiers can improve accuracy by at least 5-11\%.
\item Chapter 4: Application of Random Forests to ranking SNPs from incipient species of \emph{Anopheles} mosquitoes.  We identify multiple sources of bias that can arise and provide solutions that utilize ``out-of-the-box'' implementations of decision trees.  We provide a software implementation of our workflow using the popular machine learning library sci-kit learn, which we apply to the study of SNPs from several \emph{Anopheles} species.
\end{itemize}

We conclude by identifying topics for further investigation. \textcolor{red}{expand}
