\section{Synteny}

\subsection{Introduction}

\subsection{Methods}

\subsubsection{Data sets}


The \emph{Ae. aegypti}, \emph{An. gambiae}, \emph{L. longipalpis}, and \emph{P. papatasi} \textcolor{red}{TODO VERSIONS} peptide translations were downloaded from Vectorbase \textcolor{red}{TODO CITE}, while the \emph{D. melanogaster} and \emph{D. simulans} peptide translations were downloaded from Flybase \textcolor{red}{TODO CITE}.

\textcolor{red}{table of versions}


\subsubsection{Calculation of Scaffold Gene CDF}
The gene counts of each genome's scaffolds were normalized by dividing the gene counts by the number of genes in that organism's genome. For each genome, lists of the normalized gene counts were sorted largest to smallest and padded with 0-value entries so that all of the lists had the same length.  Cumulative sums were computed over the normalized gene counts and plotted.


\subsubsection{Generation of Dot Plots for Macrosynteny} \label{sec:synteny-methods-dotplots}
The identifiers, scaffolds, locations, and sense in the FASTA headers were extracted for each peptide sequence (Figure~\ref{fig:synteny-workflow}).  The protein IDs were cross-referenced with OrthoDB to group the proteins into ortholog groups.  Sequences without ortholog information or no orthologs in the other genomes and ortholog groups with many-to-many and one-to-many relationships were discarded.  The proteins were sorted along each scaffold by their starting coordinates, while scaffolds were ordered arbitrarily.  Scatter plots were generated by drawing dots at the positions of orthologous proteins.

\begin{figure}[H]
  \centering
  \includegraphics[width=0.5\textwidth]{figures/synteny/orthodb_dotplot_workflow}
  \caption{Workflow for Synteny}
  \label{fig:synteny-workflow}
\end{figure}

\subsubsection{Microsynteny} \label{sec:synteny-methods-synchro}
For each genome, the identifiers, protein sequences, orientations, scaffolds, and locations extracted from FASTA files for each genome and reformatted as input for the programs CHROnicle and SynChro \textcolor{red}{TODO CITE}.  SynChro ($\Delta=5$) was run on the pairs \emph{D. melanogaster} and \emph{D. simulans}, \emph{An. gambiae} and \emph{L. longipalpis}, \emph{An. gambiae} and \emph{Ae. aegypti}, and \emph{L. longipalpis} and \emph{P. papatasi}.  Synteny blocks were extracted from the \texttt{OrthBlocks synt} files.  Three-way synteny blocks for \emph{An. gambiae}, \emph{L. longipalpis}, and \emph{P. papatasi} were constructed by finding all pairs of synteny blocks for \emph{An. gambiae} and \emph{L. longipalpis} and \emph{L. longipalpis} and \emph{P. papatasi} that overlapped by at least one gene.  

The CDF for the distribution of synteny block gene counts for each pair of organisms was computed as follows: The gene counts for each synteny block were normalized by dividing the gene counts by the total number of genes found in synteny blocks. Lists of the normalized gene counts were sorted largest to smallest.  Cumulative sums were computed over the normalized gene counts. The synteny block indices were normalized by the total number of synteny blocks.

\textcolor{red}{TODO annotation of synteny blocks, distribution plots}


\subsection{Results}

\subsubsection{Genome Assembly Analysis}

\textcolor{red}{TODO DESCRIPTION}

\begin{figure}[H]
  \centering
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/genome_size_genes.pdf}
    \caption{Genome Sizes (Genes)}
  \end{subfigure}
  ~
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/scaffold_counts.pdf}
    \caption{Number of Scaffolds}
  \end{subfigure}
  ~
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/top5_scaffold_sizes.pdf}
    \caption{Top 5 Scaffold Sizes (Genes)}
  \end{subfigure}
  ~
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/gene_scaffold_cdf.pdf}
    \caption{Scaffold Genes CDF}
  \end{subfigure}
  \label{fig:scaffolds}
  \caption{}
\end{figure}

\subsubsection{Qualitative Analysis of Synteny}
Synteny between pairs of dipterans was analyzed qualitatively by generating scatter plots where each axis represents genes locations for one species and dots are drawn where orthologs occur (see Section~\ref{sec:synteny-methods-dotplots}). The presence of synteny was not detectable in a qualitative comparison of \emph{L. longipalpis} and \emph{P. papatasi} and \emph{Ae. aegypti} and \emph{A. gambiae} (Figures~\ref{fig:synteny-dotplots-sandflies}~and~\ref{fig:synteny-dotplots-mosquitoes}).  In contrast, there was extensive qualitative evidence of synteny between \emph{D. melanogaster} and \emph{D. simulans} (Figure~\ref{fig:synteny-dotplots-drosophila}).  Qualitative comparison of \emph{A. gambiae} and \emph{D. melanogaster} indicated the co-location of orthologous genes on the same chromosomes but not synteny (Figure~\ref{fig:synteny-dotplots-anopheles-drosophila}). \textcolor{red}{TODO CITE co-location}

\textcolor{red}{TODO CITE synteny of insects}


\begin{figure}[H]
  \centering
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/papatasi_longipalpis_plot}
    \caption{\emph{L. longipalpis} vs. \emph{P. papatasi}}
    \label{fig:synteny-dotplots-sandflies}
  \end{subfigure}
  ~
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/aedes_anopheles_plot}
    \caption{\emph{Ae. aegypti} vs. \emph{A. gambiae}}
    \label{fig:synteny-dotplots-mosquitoes}
  \end{subfigure}
  ~
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/dmel_dsim_plot}
    \caption{\emph{D. melanogaster} vs. \emph{D. simulans}}
    \label{fig:synteny-dotplots-drosophila}
  \end{subfigure}
  ~
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/dmel_anopheles_plot}
    \caption{\emph{A. gambiae} vs. \emph{D. melanogaster}}
    \label{fig:synteny-dotplots-anopheles-drosophila}
  \end{subfigure}
\label{fig:dot-plots}
\caption{Qualitative Analysis of Synteny}
\end{figure}

\textcolor{red}{TODO Muller elements}

\subsubsection{Quantitative Analysis of Synteny}
Synteny blocks were found between pairs of dipterans using the program SynChro (see Section~\ref{sec:synteny-methods-synchro}). Analysis of the cumulative gene densities of the synteny blocks agreed with observations made from the qualitative analysis of synteny (see Figure~\ref{fig:synteny-dist-cdf}).  Over 80\% of the \emph{D. melanogaster} and \emph{D. simulans} genes found in synteny blocks were located in 20\% of the synteny blocks.  The genes of \emph{L. longipalpis} and \emph{P. papatasi}, \emph{Ae. aegypti} and \emph{A. gambiae}, and \emph{A. gambiae} and \emph{D. melanogaster} were distributed nearly uniformally over their corresponding synteny blocks, indicating an absence of large synteny blocks.

\textcolor{red}{TODO bar plot of average, std of synteny block sizes}

\begin{figure}[H]
  \centering
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/block_counts}
    \caption{Number of Synteny Blocks}
    \label{fig:synteny-dist-counts}
  \end{subfigure}
  ~
  \begin{subfigure}[b]{0.45\textwidth}
    \includegraphics[width=\textwidth]{figures/synteny/sandflies_mosquitoes_drosophila_orth_cdf}
    \caption{Cumulative Density of Synteny Block Sizes}
    \label{fig:synteny-dist-cdf}
  \end{subfigure}
\label{fig:synteny-dist}
\caption{Quantitative Analysis of Synteny}
\end{figure}

\textcolor{red}{analysis of individual blocks}

\subsubsection{Annotation of Sand Fly Synteny Blocks}
The five largest synteny blocks for \emph{L. longipalpis} and \emph{P. papatasi} and \emph{An. gambiae}, \emph{L. longipalpis}, and \emph{P. papatasi} were extracted from the output of SynChro (see Section~\ref{sec:synteny-methods-synchro}).  The gene composition of the blocks were annotated using BLAST.

\textcolor{red}{TODO diagrams, analysis}

\subsection{Discussion and Conclusion}